src/Workspaces/Core/Portable/Shared/Extensions/SourceTextExtensions.cs

```diff
@@ -70,35 +70,64 @@ public static TextChangeRange GetEncompassingTextChangeRange(this SourceText new
         return TextChangeRange.Collapse(ranges);
     }
 
-    public static int IndexOf(this SourceText text, string value, int startIndex, bool caseSensitive)
+    public static int IndexOf(this SourceText text, string searchString, int startIndex, bool caseSensitive)
     {
-        var length = text.Length - value.Length;
-        var normalized = caseSensitive ? value : CaseInsensitiveComparison.ToLower(value);
+        if (searchString.Length == 0)
+            return -1;
+
+        var searchStringLength = searchString.Length;
+        var textEnd = text.Length - searchStringLength;
 
-        for (var i = startIndex; i <= length; i++)
+        return caseSensitive
+            ? IndexOfCaseSensitive()
+            : IndexOfCaseInsensitive();
```

Comment:
i found this cleaner as just two separate find helpers.  one for the common C# case, which needs no converting of chars, and much less branching, and one for VB. 

Both no longer alloc.  Only the VB one has some special complex logic around case insensitivity.